README.md

```diff
@@ -56,44 +61,93 @@ public function registerBlocks(): array
 
+For example, let's say you have a **Title** block which can display a heading tag in your content. You may optionally want to align it to left, center or right, and define which heading tag to use. The best practice would be to have a `content` field in the `fields` definition, because it's the actual content being displayed. The `alignment` and `tag` would become part of the `config` configuration.
+
+**Example:**
+
+```
+name: Title
+description: Adds a title
+icon: icon-heading
+context: ["content"]
+fields:
+    content:
+        label: false
+        span: full
+        type: text
+config:
+    size:
+        label: Size
+        span: auto
+        type: dropdown
+        default: h2
+        options:
+            h1: H1
+            h2: H2
+            h3: H3
+            h4: H4
+            h5: H5
+    alignment_x:
+        label: Alignment
+        span: auto
+        type: dropdown
+        default: center
+        options:
+            left: Left
+            center: Centre
+            right: Right
+==
+{% if config.alignment_x == 'left' %}
+    {% set alignment = 'text-left' %}
+{% elseif config.alignment_x == 'center' or not config.alignment_x %}
+    {% set alignment = 'text-center' %}
+{% elseif config.alignment_x == 'right' %}
+    {% set alignment = 'text-right' %}
+{% endif %}
+
+<{{ config.size|default('h2') }} class="{{ alignment }}">
+    {{ content }}
+</{{ config.size|default('h2') }}>
+```
+
+> **NOTE:** With `config` options, these must always be treated as "optional" due to limitations with the Inspector widget. The Inpsector widget only populates values in the block when they are changed. Always have a fallback in your Twig content.
 
 ## Using the `blocks` FormWidget
 
 In order to provide an interface for managing block-based content, this plugin provides the `blocks` FormWidget. This widget can be used in the backend as a form field to manage blocks.
 
 The `blocks` FormWidget supports two additional properties:
 
-- `allow`: An array of block types that are allowed to be added to the widget. If specified, only those block types listed will be available to add to the current instance of the field.
-- `ignore`: A list of block types that are not allowed to be added to the widget. If specified, all block types will be available to add to the current instance of the field, except those listed.
+- `allow`: An array of block types that are allowed to be added to the widget. If specified, only those block types listed will be available to add to the current instance of the field. You can specify either the name of a block or an entire context to allow.
+- `ignore`: A list of block types that are not allowed to be added to the widget. If not specified, all block types will be available to add to the current instance of the field. You can specify either the name of a block or an entire context to ignore.
```

Comment:
I think if we explain that `allow` and `ignore` accept both individual block names and contexts, it might assist people with being specific about their naming. If you reckon that it'll trip people up, we could always split it to `allowContexts` / `ignoreContexts` and `allowBlocks` / `ignoreBlocks`.

Reply:
Switched `context` terminology to `tags`.

I've also updated the `allow` and `ignore` schema to allow the following:

### A list of blocks to `allow` (or `ignore`):

```yaml
allow:
    - title
    - text
    - image
```

### A list of blocks and tags:

```yaml
allow:
    blocks:
        - image
    tags:
        - textual
        - buttons
```